masq/src/non_interactive_mode.rs

```diff
@@ -29,24 +31,34 @@ impl Default for Main {
 impl Main {
     pub fn new() -> Self {
         Self {
-            command_factory: Box::new(CommandFactoryReal::new()),
-            processor_factory: Box::new(CommandProcessorFactoryReal {}),
+            non_interactive_clap_factory: Box::new(NonInteractiveClapFactoryReal),
+            command_factory: Box::new(CommandFactoryReal),
+            processor_factory: Box::new(CommandProcessorFactoryReal),
         }
     }
 
     fn extract_subcommand(args: &[String]) -> Option<Vec<String>> {
-        let args_vec: Vec<String> = args.to_vec();
-        for idx in 1..args_vec.len() {
-            let one = &args_vec[idx - 1];
-            let two = &args_vec[idx];
-            if !one.starts_with("--") && !two.starts_with("--")
-                || two.eq("--help")
-                || two.eq("--version")
-            {
-                return Some(args_vec.into_iter().skip(idx).collect());
-            }
+        if args.len().eq(&1) {
+            return None;
+        }
+        let vec_candidate = args
+            .to_vec()
+            .into_iter()
+            .skip(1 + Self::eliminate_pre_subcommand_items(args[1].as_str()))
+            .collect::<Vec<String>>();
+        if !vec_candidate.is_empty() {
```

Comment:
Can you reverse the sense of the check here to eliminate the bang? Bangs can be difficult to read, and reversing this one is pretty easy.